File: frontend/features/practice-slots/types/parts.ts
export interface Part {
  id?: string;
  name: string;
  display_name: string;
  description?: string;
  is_active: boolean;
  sort_order: number;
  created_at?: string;
  updated_at?: string;
}

export interface PartCreate {
  name: string;
  display_name: string;
  description?: string;
  is_active?: boolean;
  sort_order?: number;
}

export interface PartUpdate {
  name?: string;
  display_name?: string;
  description?: string;
  is_active?: boolean;
  sort_order?: number;
}

File: backend/app/schemas/parts.py
from datetime import datetime
from typing import Optional
from uuid import UUID
from pydantic import BaseModel, ConfigDict


class PartBase(BaseModel):
    name: str
    display_name: str
    description: Optional[str] = None
    is_active: bool = True
    sort_order: int = 0


class PartCreate(PartBase):
    pass


class PartUpdate(BaseModel):
    name: Optional[str] = None
    display_name: Optional[str] = None
    description: Optional[str] = None
    is_active: Optional[bool] = None
    sort_order: Optional[int] = None


class PartResponse(PartBase):
    id: UUID
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    model_config = ConfigDict(from_attributes=True)
